servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DefaultDnsClient.java

```diff
@@ -348,8 +351,14 @@ protected Future<DnsAnswer<HostAndPort>> doDnsQuery() {
                                                 throw new IllegalArgumentException(
                                                         "Unsupported DNS record type for SRV query: " + dnsRecord);
                                             }
-                                            if (dnsRecord.timeToLive() < minTTLSeconds) {
-                                                minTTLSeconds = dnsRecord.timeToLive();
+                                            if (dnsRecord.timeToLive() < answerTTL) {
+                                                answerTTL = dnsRecord.timeToLive();
+                                            }
+                                            if (answerTTL > maxTTL) {
+                                                answerTTL = maxTTL;
+                                                LOGGER.info("DNS Record {} has a high TTL of {}s which is " +
```

Comment:
1. There is a chance that DNS can return different TTL values for different records, but in practice they are usually the same. This for loop tries to identity the lowest ttl value for all records. Consider moving this out of the loop to log only once after we identified the effective answerTTL/minTTLSeconds, no need to bother with every record.
2. Having it here only covers SRV resolutions, for A/AAAA resolutions there is a different subscription impl. To cover all of them, consider moving this logging to `handleResolveDone0` where we schedule the next query.